src/common.c

```diff
@@ -0,0 +1,31 @@
+#include "common.h"
+
+/**
+ * @brief Read a status flag.
+ *
+ * @param[in] adc_peripheral Unsigned int32. ADC register address base @ref
+ * adc_reg_base
+ * @param[in] flag Unsigned int32. Status register flag  @ref ADC_SR values.
+ * @returns boolean: flag set.
+ */
```

Comment:
I would not include the parameter type in the docstring, as it is clearly declared in the function (we are not declaring it in any other docstrings I think).

If `@ref` are supposed to be references, then I would either use `(see <whatever)` if it is in the same line or create a new line with `@see <whatever>.`, as we have done in other docstrings.

Reply:
Fixed, I removed the parameter type in the docstring, included an extra line between @XXXX expressions and changed @ref by @see.